.github/workflows/ci-vulnerability-scans.yml

```diff
@@ -0,0 +1,81 @@
+# This file is an example CI test pipeline that will run vulnerability scans on your image
+# to ensure images built are secure before they are deployed. 
+name: API Linting & Testing CI
+
+on:
+  push:
+    branches:
+      - main
+  pull_request:
+
+jobs:
+  check-for-changes:
+    runs-on: ubuntu-latest
+    # Set job outputs to values from filter step
+    outputs:
+      needs-testing: ${{ steps.filter.outputs.needs-testing }}
+    steps:
+      - uses: actions/checkout@v3
+
+      - uses: dorny/paths-filter@v2
+        id: filter
+        with:
+          filters: |
+            needs-testing:
+              - app/**
+              - .github/**/*.yml
```

Comment:
I bias towards simplicity, since I don't think the gain in runtime is worth the extra job and added dependency on a third party action. Since we're removing testing/linting from this workflow, I think it can be safe to say that if there are any changes in app/** we'd want to re-run scans, and specific projects can even narrow that down further e.g. if they want to narrow it to app/Dockerfile + app/src/**, but in template-infra we can't do that because we can't assume anything about the directory structure within app/